File: fieldsync-inventory-backend/src/main/java/com/example/fieldsync_inventory_backend/entity/inventory/RiceVarietyEntity.java
package com.example.fieldsync_inventory_backend.entity.inventory;

import jakarta.persistence.*;
import jakarta.persistence.Table;
import lombok.*;
import com.example.fieldsync_inventory_backend.entity.Auditable;

import java.time.Instant;

@Entity
@Table(name = "rice_variety")
@Getter
@Setter
@ToString(callSuper = true)
@EqualsAndHashCode(callSuper = true)
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class RiceVarietyEntity extends Auditable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(length = 50, unique = true, nullable = false)
    private String name;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(name="image_url", columnDefinition = "TEXT")
    private String imageUrl;

    @Column(name = "deleted_at")
    private Instant deletedAt;
}

File: fieldsync-inventory-app/app/src/main/java/com/example/fieldsync_inventory_app/data/local/entity/RiceVarietyEntity.kt
package com.example.fieldsync_inventory_app.data.local.entity

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "rice_variety")
data class RiceVarietyEntity(
    @PrimaryKey val id: Int, // Use the ID from the backend as the primary key
    val name: String,
    @ColumnInfo(name = "image_url")
    val imageUrl: String?,
    val description: String,
    @ColumnInfo(name = "created_at")
    val createdAt: Long,
    @ColumnInfo(name = "updated_at")
    val updatedAt: Long,
    @ColumnInfo(name = "deleted_at")
    val deletedAt: Long?
)